scipy/optimize/_bracket.py

```diff
@@ -450,10 +441,12 @@ def _bracket_minimum_iv(func, xm0, xl0, xr0, xmin, xmax, factor, args, maxiter):
         xr0_not_supplied = True
 
     factor = 2.0 if factor is None else factor
-    xl0, xm0, xr0, xmin, xmax, factor = xp.broadcast_arrays(
-        xp.asarray(xl0), xm0, xp.asarray(xr0), xp.asarray(xmin),
-        xp.asarray(xmax), xp.asarray(factor)
-    )
+
+    xm0, xl0, xr0, xmin, xmax, factor = xp_promote(
+        xm0, xl0, xr0, xmin, xmax, factor, broadcast=True, force_floating=True, xp=xp)
+
+    if not xp.isdtype(xm0.dtype, ('integral', 'real floating')):
+        raise ValueError('`xm0` must be numeric and real.')
```

Comment:
Could you update the checks on lines 451:468? Probably a good idea to make it into a for loop.

Reply:
Not here.